io-engine/src/subsys/registration/registration_grpc.rs

```diff
@@ -19,6 +19,27 @@ const HTTP_KEEP_ALIVE_INTERVAL: Duration = Duration::from_secs(10);
 /// The http2 keep alive TIMEOUT.
 const HTTP_KEEP_ALIVE_TIMEOUT: Duration = Duration::from_secs(20);
 
+#[derive(Clone, Debug, PartialEq)]
+/// ApiVersion to be supported
+pub enum ApiVersion {
+    /// V0 Version of api
+    V0,
+    /// V1 version of api
+    V1,
+}
+
+impl FromStr for ApiVersion {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
```

Comment:
s.to_lower() and then can simply check "v0" and "v1" ?

Reply:
Ahh right.